src/features/messages/cards/TransactionCard.tsx

```diff
@@ -207,10 +208,15 @@ function DeliveryStatus({ children }: PropsWithChildren<unknown>) {
   );
 }
 
-function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
+function CallDataModal({ debugResult,chainId }: { debugResult?: MessageDebugResult,chainId:ChainId }) {
   const [isOpen, setIsOpen] = useState(false);
+  const disabled=false
+  const [buttonText,setButtonText]=useState<string>("Simulate Handle Call")
   if (!debugResult?.calldataDetails) return null;
   const { contract, handleCalldata } = debugResult.calldataDetails;
+  function handleClick() {
```

Comment:
event handlers (like click) should use lambda (arrow fn) syntax to be bound correctly.